src/coreComponents/constitutive/fluid/PVTDriver.cpp

```diff
@@ -0,0 +1,332 @@
+    GEOSX_LOG_RANK_0( "  Output ................. " << m_outputFile );
+    GEOSX_LOG_RANK_0( "  Baseline ............... " << m_baselineFile );
+  }
+
+  // create a dummy discretization with one quadrature point for
+  // storing constitutive data
+
+  conduit::Node node;
+  dataRepository::Group rootGroup( "root", node );
+  dataRepository::Group discretization( "discretization", &rootGroup );
+
+  discretization.resize( 1 );   // one element
+  baseFluid.allocateConstitutiveData( discretization, 1 );   // one quadrature point
+
+  // pass the solid through the ConstitutivePassThru to downcast from the
+  // base type to a known model type.  the lambda here then executes the
+  // appropriate test driver. note that these calls will move data to device if available.
+
+  constitutiveUpdatePassThru( baseFluid, [&] ( auto & selectedFluid )
+  {
+    using FLUID_TYPE = TYPEOFREF( selectedFluid );
+    runTest< FLUID_TYPE >( selectedFluid, m_table );
+  } );
+
+  // move table back to host for output
+  m_table.move( LvArray::MemorySpace::host );
+
+  if( m_outputFile != "none" )
+  {
+    outputResults();
+  }
+
+  if( m_baselineFile != "none" )
+  {
+    compareWithBaseline();
+  }
+
+  return false;
+}
+
+
+template< typename FLUID_TYPE >
+void PVTDriver::runTest( FLUID_TYPE & fluid, arrayView2d< real64 > & table )
+{
+  // get number of phases and components
+
+  localIndex const NC = fluid.numFluidComponents();
+  localIndex const NP = fluid.numFluidPhases();
+
+  // prefer output in mass
+
+  fluid.setMassFlag( true );
+
+  // get output data views
+
+  arrayView2d< real64 const, multifluid::USD_FLUID > totalDensity   = fluid.totalDensity();
+  arrayView3d< real64 const, multifluid::USD_PHASE > phaseFraction  = fluid.phaseFraction();
+  arrayView3d< real64 const, multifluid::USD_PHASE > phaseDensity   = fluid.phaseDensity();
+  arrayView3d< real64 const, multifluid::USD_PHASE > phaseViscosity = fluid.phaseViscosity();
+
+  // create kernel wrapper
+
+  typename FLUID_TYPE::KernelWrapper kernelWrapper = fluid.createKernelWrapper();
+
+  // set composition to user specified feed
+  // it is more convenient to provide input in molar, so perform molar to mass conversion here
+
+  GEOSX_ASSERT_EQ( NC, m_feed.size() );
+  array2d< real64, compflow::LAYOUT_COMP > compositionValues( 1, NC );
+
+  real64 sum = 0.0;
+  for( localIndex i = 0; i < NC; ++i )
+  {
+    compositionValues[0][i] = m_feed[i] * fluid.componentMolarWeights()[i];
+    sum += compositionValues[0][i];
+  }
+  for( localIndex i = 0; i < NC; ++i )
+  {
+    compositionValues[0][i] /= sum;
+  }
+
+  arraySlice1d< real64 const, compflow::USD_COMP - 1 > const composition = compositionValues[0];
+
+  // perform fluid update using table (P,T) and save resulting total density, etc.
+  // note: column indexing should be kept consistent with output file header below.
+
+  forAll< parallelDevicePolicy<> >( 1, [=]  GEOSX_HOST_DEVICE ( integer const ei )
```

Comment:
```suggestion 
  using ExecPolicy = typename FLUID_TYPE::exec_policy;
  forAll< ExecPolicy >( 1, [=]  GEOSX_HOST_DEVICE ( integer const ei )
```
This is what we do in the flow solvers since PVTPackage is not ready for the GPU, and its `compute` function is disabled if used with `parallelDevicePolicy` on a GPU platform (see [here](https://github.com/GEOSX/GEOSX/blob/0ebaf87e3f7a4b171e1b0e03d5213d75c91aa31d/src/coreComponents/constitutive/fluid/CompositionalMultiphaseFluid.hpp#L163-L164)). So using `parallelDevicePolicy` here may result in a failed unit test on Lassen if `CompositionalMultiphaseFluid` is selected (to be confirmed).